### PR TITLE
Add "spack versions --new" flag to only show new versions

### PR DESCRIPTION
**Pull request by vvolkl**
Example (using a package from https://github.com/key4hep/key4hep-spack, just because its version history is a nice illustration):

```
~$ spack versions marlin
==> Safe versions (already checksummed):
  master  1.17
==> Remote versions (not yet checksummed):
  01-17-01  01-16  01-15-02  01-15-01  01-15  01-14  01-13  01-12  01-11  01-05
```
```
~$ spack versions --new marlin
==> Safe versions (already checksummed):
  master  1.17
==> Remote versions (not yet checksummed):
  01-17-01
```

The reasoning behind this is that a lot of packages have old versions that are outdated or not checksummed in the recipe for other reasons. As a maintainer, I'm mostly interested in checking for new versions - all the versions older than 01-17  in my example are basically noise. Together with `spack maintainers --by-user` this can then also be used to create automated checks, reminding me when new versions are available (not all of the packages I'm maintaining are on github, where I could configure notifications).

### CODE REVIEW

**Comment by vvolkl** [comment]
Just realizing that `--latest` could be understood to show only *the* latest version. Maybe better to name it `--new`?

**Comment by vvolkl** [comment]
For scripting, I think it would also be nice if the only thing this command writes to sys.stdout are the remote versions (right now it is both safe and remote versions). So that
```
export versions=$(spack versions --latest marlin)
```
gives 
```
echo $versions
01-17-01
```
and not
```
echo $versions
master  1.17 01-17-01
```
@tgamblin @adamjstewart @alalazo Is it possible to pass `output=tty.debug` to `colify`, or what would be the best way to do this?

**Comment by adamjstewart** [comment]
> For scripting, I think it would also be nice if the only thing this command writes to sys.stdout are the remote versions (right now it is both safe and remote versions).

We have a `--safe-only` flag, shouldn't be hard to add a `--remote-only` flag (and make them mutually exclusive).

File: lib/spack/spack/cmd/versions.py
Comment (alalazo) [inline lib/spack/spack/cmd/versions.py]: For the sake of being consistent, can we either use:
```
--safe, --remote, --new
```
or alternatively:
```
--safe-only, --remote-only, --new-only
```
instead of having the `-only` suffix only for the first two options and not for the last?

Comment (vvolkl) [reply lib/spack/spack/cmd/versions.py]: Yes, I can do that. Would prefer the former (less typing) but on the other hand that would change the existing `--safe-only` argument. Any opinions on this?

I didn't add `--new` to the output group initially, because the code logic seemed simpler, as  it still needs to fetch the safe versions to compare. Also you won't get the printout of both safe and new remote versions like this. But the arguments are more consistent with your suggestion, which is always nice.

Comment (vvolkl) [reply lib/spack/spack/cmd/versions.py]: Wait, or did you mean only renaming `--new` to `--new-only` without making it mutually exclusive with `--remote-only`?

Comment (adamjstewart) [reply lib/spack/spack/cmd/versions.py]: I vote the shorter flags. If you want to be backwards compatible, you can keep safe-only and add a deprecation warning until the next release. All should be mutually exclusive.

Comment (vvolkl) [reply lib/spack/spack/cmd/versions.py]: @adamjstewart  sounds good - done

**Comment by adamjstewart** [comment]
Can you add a test for `--new`?

**Comment by vvolkl** [comment]
@adamjstewart Good question. I'm not sure there is a package that consistently has newer remote versions than checksummed. I could create a mock package, or is it fine not to check the output? Because if there are no remote versions the command will not return a "success" return value, that's why I removed the `--new` test.

**Comment by adamjstewart** [comment]
Ah, I see. Maybe @alalazo would know if it's possible to ignore the return code so we at least get coverage.

**Comment by vvolkl** [comment]
@adamjstewart @alalazo Added a mock package to  test `spack versions --new`. Should be good to merge -- Let me know if there is anything else missing.

File: lib/spack/spack/cmd/versions.py
Comment (alalazo) [inline lib/spack/spack/cmd/versions.py]: :+1:

File: lib/spack/spack/cmd/versions.py
Comment (alalazo) [inline lib/spack/spack/cmd/versions.py]:
```suggestion
                        help='[deprecated] only list safe versions '
                        'of the package')
```

File: lib/spack/spack/cmd/versions.py
Comment (alalazo) [inline lib/spack/spack/cmd/versions.py]:
```suggestion
                        help='only list remote versions newer than '
                        'the latest checksummed version')
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Only two very minor typos (lack of spaces between words) Otherwise LGTM.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@adamjstewart Waiting for your green light to merge

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
